Connext_DDS_QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,108 @@
+# **External Dependency Quality declaration** `RTI-Connext-DDS`
+
+This document is a declaration of software quality for the `RTI-Connext-DDS` external dependency, based on the guidelines in [REP-2004](https://github.com/ros-infrastructure/rep/blob/master/rep-2004.rst).
+
+As stated in their core documentation, *RTI® Connext® DDS solutions provide a flexible connectivity software framework for integrating data sources of all types (...) It connects data within applications as well as across devices, systems and networks. _Connext DDS_ also delivers large data sets with microsecond performance and granular quality-of-service control.* 
+
+This Quality Declaration claims that the external dependency `RTI-Connext-DDS` is in the Quality Level 4 category.
+
+Below are the rationales, notes, and caveats for this claim, organized by each requirement listed in the [Package Requirements for Quality Level 4 in REP-2004](https://www.ros.org/reps/rep-2004.html).
+
+## Summary
+`RTI-Connext-DDS` meets most of the standards defined for Quality Level 1, including broad documentation and testing of its features and API.
+
+As this package is not open-source, many policies related to quality control are not openly documented. For example, it is difficult to find information relating to its versioning policy. For ROS 2 distributions, packages may need to rely on strict versions, like `5.3.1`, to ensure API/ABI stability. 
+
+There are no clear public reports of code coverage, so it is not clear how much of their code is tested. However, they do describe their approach to testing in a [blog entry](https://www.rti.com/blog/software-testing-at-rti). 
+
+There is a list of [known academic projects](https://community.rti.com/projects) using RTI Connext DDS and also they have published a [white paper](https://www.rti.com/whitepapers/how-to-achieve-production-grade-deployment-with-ros-2-and-rti-connext-dds) with rationale about how to implement production-grade deployments using their software in conjunction with ROS 2.
+
+In terms of ROS 2 package metrics this library is considered to be Quality Level 4. Adding unit testing for the functions used in ROS 2 packages, coverage statistics and version pinning will be needed to achieve Quality Level 1.
```

Comment:
It should be made clear who needs to do what to increase the quality level. RTI doesn't really have to add unit testing, or pin versions. Instead, they just need to be more transparent about their code coverage, API coverage, versioning policy, etc in order for consumers of the package to better understand how to depend on it.